extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,239 @@
+                "url": "http://localhost:8888/api/packagemanager/packages/check_update?project=" + project,
+                "method": "GET",
+                "headers": {
+                    "Content-Type": "application/json",
+                    "cache-control": "no-cache"
+                },
+                "processData": false,
+                "data": ""
+            };
+
+            return new Promise(resolve => {
+                $.ajax(settings).done(function (response) {
+                    resolve(response);
+                });
+            });
+        },
+
+        checknewversion: function (updates, packagename, pkgversion) {
+            for (let element of updates) {
+                if (element.name === packagename) {
+                    return element.version;
+                }
+            }
+            return pkgversion;
+        },
+
+        load: async function () {
+            let update = this.update;
+            let checkupdate = this.checkupdate;
+            let checknewversion = this.checknewversion;
+            jQuery(async function () {
+                $('#updatebtn').unbind();
+                $('#updatebtn').click(async function () {
+                    let selectedPackages = sessionStorage.getItem("selectedPackages");
+                    if (null === selectedPackages)
+                        selectedPackages = [];
+                    else {
+                        selectedPackages = selectedPackages.split(',');
+                        if (selectedPackages[0].length < 1) selectedPackages = [];
+                    }
+                    let project = sessionStorage.getItem("project");
+                    let html = "<p> The following packages are about to be updated: </p> </br>";
+                    let resp = await checkupdate(project);
+                    let updates = resp.updates;
+                    let packages = [];
+                    let ct = 0;
+                    if (selectedPackages.length > 0) {
+                        html += "<ul>";
+                        selectedPackages.forEach(element => {
+                            element = element.split('=');
+                            let pkg = element[0];
+                            let ver = element[1];
+                            let nver = checknewversion(updates, pkg, ver);
+                            if (nver != ver) {
+                                packages.push(pkg);
+                                html += "<li>";
+                                html += pkg + "  ";
+                                html += ver + " -> ";
+                                html += nver;
+                                html += "</li>";
+                                ct += 1;
+                            }
+                        });
+                        html += "</ul>";
+                    }
+                    else {
+                        var list = document.getElementsByClassName("installed-values");
```

Comment:
if you're using jquery, why don't you do `$(".installed-values")` instead of `getElementsByClassName`?